Orange/data/pandas_compat.py

```diff
@@ -281,13 +298,19 @@ def table_from_frame(df, *, force_nominal=False):
     XYM, domain = vars_from_df(df, force_nominal=force_nominal)
 
     if hasattr(df, 'orange_weights') and hasattr(df, 'orange_attributes'):
-        W = [df.orange_weights[i] for i in df.index
-             if i in df.orange_weights]
+        W = [df.orange_weights[i] for i in df.index if i in df.orange_weights]
         if len(W) != len(df.index):
             W = None
         attributes = df.orange_attributes
-        ids = [int(i[2:]) if str(i).startswith('_o') else Table.new_id()
-               for i in df.index]
+        if isinstance(df.index, pd.MultiIndex) or not is_string_dtype(df.index):
+            # we can skip checking for Orange indices when MultiIndex an when
+            # not string dtype and so speedup the conversion
+            ids = None
+        else:
+            ids = [
+                int(i[2:]) if str(i).startswith("_o") else Table.new_id()
```

Comment:
Is it anyhow assured that `i[2:]` can always cast to `int`?

Reply:
You are right. There is no guarantee for that in case the index with `_o` was not produced with the transformation of Table to frame. It should be fixed now.